lib/matplotlib/legend.py

```diff
@@ -981,8 +981,7 @@ def _auto_legend_data(self):
             elif isinstance(artist, Collection):
                 transform, transOffset, hoffsets, _ = artist._prepare_points()
                 if len(hoffsets):
-                    for offset in transOffset.transform(hoffsets):
-                        offsets.append(offset)
+                    offsets.extend(offset for offset in transOffset.transform(hoffsets))
```

Comment:
Not near a dev environment, but can't we just do
```suggestion
                    offsets.extend(transOffset.transform(hoffsets))
```

Reply:
Good thinking! I'll check.